hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/webapp/dao/RMQueueAclInfo.java

```diff
@@ -25,9 +25,10 @@
 @XmlRootElement
 @XmlAccessorType(XmlAccessType.FIELD)
 public class RMQueueAclInfo {
-  protected boolean allowed;
+  protected Boolean allowed;
```

Comment:
why not boolean?

Reply:
In Federation mode, we may have multiple SubClusters, and to verify whether the user has queue permissions, multiple results may be returned, and these results cannot be directly integrated.


We directly call the RM interface, and the following return results can be obtained

```
http://hostname:8088/ws/v1/cluster/queues/default/access?user=yarn

<rmQueueAclInfo>
  <allowed>true</allowed>
  <user>yarn</user>
  <diagnostics/>
</rmQueueAclInfo>
```

We visit the router and hope to get the following results:

```
http://hostname:8089/ws/v1/cluster/queues/default/access?user=yarn

<federationRMQueueAclInfo>
  <subCluster>
    <allowed>true</allowed>
    <user>yarn</user>
    <diagnostics/>
    <subClusterId>SC-1</subClusterId>
  </subCluster>
  <subCluster>
    <allowed>true</allowed>
    <user>yarn</user>
    <diagnostics/>
    <subClusterId>SC-2</subClusterId>
  </subCluster>
</federationRMQueueAclInfo>
```
I found that if allowed is a boolean type, the return result is like this, Because the XmlRootElement displays the field, if the field is null, it will not be displayed in the result, I upgraded allowed from boolean to boolean

```
<federationRMQueueAclInfo>
  <allowed>false</allowed> 
  <subCluster>
    <allowed>true</allowed>
    <user>yarn</user>
    <diagnostics/>
    <subClusterId>SC-1</subClusterId>
  </subCluster>
  <subCluster>
    <allowed>true</allowed>
    <user>yarn</user>
    <diagnostics/>
    <subClusterId>SC-2</subClusterId>
  </subCluster>
</federationRMQueueAclInfo>
```